fw/http_parser.c

```diff
@@ -3295,15 +3295,26 @@ __req_parse_x_tempesta_cache(TfwHttpMsg *hm, unsigned char *data, size_t len)
 		/* "X-Tempesta-Cache" ":" method */
 		TRY_STR_fixup(&TFW_STR_STRING("get"), Req_I_X_Tempesta_Cache,
 			      I_Tempesta_Cache_get);
-		return CSTR_NEQ;
+		/* If the method is not "GET" just ignore the rest of the
+		 * header line. */
+		__FSM_I_JMP(I_Tempesta_Cache_skip);
 	}
 
 	__FSM_STATE(I_Tempesta_Cache_get) {
 		if (likely(IS_CRLF(c))) {
 			__set_bit(TFW_HTTP_B_PURGE_GET, hm->flags);
 			return __data_off(p);
 		}
-		return CSTR_NEQ;
+		__FSM_I_JMP(I_Tempesta_Cache_skip);
+	}
+
+	__FSM_STATE(I_Tempesta_Cache_skip) {
+		parser->_acc = 0;
+		/* Skip the rest of the line. */
+		__FSM_I_MATCH_MOVE(nctl, I_Tempesta_Cache_skip);
+		if (!IS_CRLF(*(p + __fsm_sz)))
```

Comment:
Sorry, I'm super confused now. Judging by the suggested test, we want to parse a comma delimited list of X-Tempesta-Cache methods now?

What about something like `X-Tempesta-Cache: head, get` should this work too?

Reply:
I've added these tests. It actually just works as long as the method list starts with "get".

Reply:
I wrote "like", not "exactly" :) We may allow or not `get, get` and we do not need to support `head, get`. Since `get , ` isn't correct value and we do not want to overcomplicate parser for senseless values, it makes sense to prohibit `get, get`